fastlane/lib/fastlane/runner.rb

```diff
@@ -258,13 +279,30 @@ def execute_action(method_sym, class_ref, arguments, custom_dir: nil, from_actio
         raise e
       rescue FastlaneCore::Interface::FastlaneError => e # user_error!
         FastlaneCore::CrashReporter.report_crash(exception: e)
-        collector.did_raise_error(method_sym) if e.fastlane_should_report_metrics?
+        if e.fastlane_should_report_metrics?
```

Comment:
same here, could combine into a function.
Might be worth creating a "metics session manager" that can contain the reusable functions I mentioned above

Reply:
I'm not 💯 convinced a metrics session manager is a good solution, but just thinking outloud 😄